### PR TITLE
chore: Release automation

### PR DESCRIPTION
**Pull request by vlad-yakovlev**
**Before merge we need to add GH_TOKEN and NPM_TOKEN to Secrets**

**v0.1.0 will be automatically published after merge**

Added release automation (#185)

I've `dry-run`-ed parts of this action for test:

<details>
  <summary>semantic-release dry-run</summary>
  
```
[3:28:05 AM] [semantic-release] › ℹ  Running semantic-release version 19.0.3
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/github"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[3:28:05 AM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[3:28:12 AM] [semantic-release] › ⚠  Run automated release from branch master on repository git@github.com:vlad-yakovlev/remeda.git in dry-run mode
[3:28:14 AM] [semantic-release] › ✔  Allowed to push to the Git repository
[3:28:14 AM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/npm"
[3:28:14 AM] [semantic-release] [@semantic-release/npm] › ℹ  Verify authentication for registry https://registry.npmjs.org/
[3:28:14 AM] [semantic-release] [@semantic-release/npm] › ℹ  Wrote NPM_TOKEN to /private/var/folders/mt/z54x8v_55yx0hq_6n2hp86_c0000gn/T/315c6fcad0fb700e24aa57f91fd94558/.npmrc
vlad-yakovlev
[3:28:15 AM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[3:28:15 AM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[3:28:15 AM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[3:28:16 AM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[3:28:16 AM] [semantic-release] › ℹ  Found git tag v0.0.35 associated with version 0.0.35 on branch master
[3:28:16 AM] [semantic-release] › ℹ  Found 13 commits since last release
[3:28:16 AM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: add pr template (#183)
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Add uniqWith (#181)

* Add uniqWith

* Fix tests & examples

* Add uniqWith to index
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Bump jsdom from 16.4.0 to 16.7.0 (#176)

Bumps [jsdom](https://github.com/jsdom/jsdom) from 16.4.0 to 16.7.0.
- [Release notes](https://github.com/jsdom/jsdom/releases)
- [Changelog](https://github.com/jsdom/jsdom/blob/master/Changelog.md)
- [Commits](https://github.com/jsdom/jsdom/compare/16.4.0...16.7.0)

---
updated-dependencies:
- dependency-name: jsdom
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Bump minimist from 1.2.0 to 1.2.6 in /docs (#165)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.0 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.0...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Bump minimist from 1.2.5 to 1.2.6 (#164)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Bump chownr from 1.0.1 to 1.1.4 in /docs (#160)

Bumps [chownr](https://github.com/isaacs/chownr) from 1.0.1 to 1.1.4.
- [Release notes](https://github.com/isaacs/chownr/releases)
- [Commits](https://github.com/isaacs/chownr/compare/v1.0.1...v1.1.4)

---
updated-dependencies:
- dependency-name: chownr
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Bump ansi-regex from 5.0.0 to 5.0.1 (#150)

Bumps [ansi-regex](https://github.com/chalk/ansi-regex) from 5.0.0 to 5.0.1.
- [Release notes](https://github.com/chalk/ansi-regex/releases)
- [Commits](https://github.com/chalk/ansi-regex/compare/v5.0.0...v5.0.1)

---
updated-dependencies:
- dependency-name: ansi-regex
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Bump tmpl from 1.0.4 to 1.0.5 (#148)

Bumps [tmpl](https://github.com/daaku/nodejs-tmpl) from 1.0.4 to 1.0.5.
- [Release notes](https://github.com/daaku/nodejs-tmpl/releases)
- [Commits](https://github.com/daaku/nodejs-tmpl/commits/v1.0.5)

---
updated-dependencies:
- dependency-name: tmpl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Bump path-parse from 1.0.5 to 1.0.7 in /docs (#142)

Bumps [path-parse](https://github.com/jbgutierrez/path-parse) from 1.0.5 to 1.0.7.
- [Release notes](https://github.com/jbgutierrez/path-parse/releases)
- [Commits](https://github.com/jbgutierrez/path-parse/commits/v1.0.7)

---
updated-dependencies:
- dependency-name: path-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Bump path-parse from 1.0.5 to 1.0.7 (#141)

Bumps [path-parse](https://github.com/jbgutierrez/path-parse) from 1.0.5 to 1.0.7.
- [Release notes](https://github.com/jbgutierrez/path-parse/releases)
- [Commits](https://github.com/jbgutierrez/path-parse/commits/v1.0.7)

---
updated-dependencies:
- dependency-name: path-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Add differenceWith to index (#182)
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Add differenceWith (#180)

* Add differenceWith

* Replace find with every

* Fix differenceWith tests & examples
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Add pickBy (#178)
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[3:28:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 13 commits complete: minor release
[3:28:16 AM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[3:28:16 AM] [semantic-release] › ℹ  The next release version is 0.1.0
[3:28:16 AM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[3:28:16 AM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[3:28:16 AM] [semantic-release] › ⚠  Skip step "prepare" of plugin "@semantic-release/npm" in dry-run mode
[3:28:16 AM] [semantic-release] › ⚠  Skip v0.1.0 tag creation in dry-run mode
[3:28:16 AM] [semantic-release] › ⚠  Skip step "publish" of plugin "@semantic-release/npm" in dry-run mode
[3:28:16 AM] [semantic-release] › ⚠  Skip step "publish" of plugin "@semantic-release/github" in dry-run mode
[3:28:16 AM] [semantic-release] › ⚠  Skip step "success" of plugin "@semantic-release/github" in dry-run mode
[3:28:16 AM] [semantic-release] › ✔  Published release 0.1.0 on default channel
[3:28:16 AM] [semantic-release] › ℹ  Release note for version 0.1.0:
# 0.1.0 (https://github.com/vlad-yakovlev/remeda/compare/v0.0.35...v0.1.0) (2022-07-16)

### Features

    * Add differenceWith (#180 (https://github.com/vlad-yakovlev/remeda/issues/180)) (ade8ef7 (https://github.com/vlad-yakovlev/remeda/commit/ade8ef7e14f37987547e823a0f89b32f8b7bb0ab))
    * Add pickBy (#178 (https://github.com/vlad-yakovlev/remeda/issues/178)) (f7a1974 (https://github.com/vlad-yakovlev/remeda/commit/f7a197419397e4954e39f0c7548ad575def2ec80))
    * Add uniqWith (#181 (https://github.com/vlad-yakovlev/remeda/issues/181)) (8e2999b (https://github.com/vlad-yakovlev/remeda/commit/8e2999b88a88888d5523181444972ced870635dc))
```
</details>

---

Make sure that you:

- [x] Typedoc added for new methods and updated for changed
- [x] Tests added for new methods and updated for changed
- [x] New methods added to `src/index.ts`
- [x] New methods added to `mapping.md`

### CODE REVIEW

File: .github/workflow/release.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+name: Release
+
+on:
+  push:
+    branches:
+      - master
```
Comment (TkDodo) [inline .github/workflow/release.yml:7]: can we take this opportunity to rename `master` to `main` ?

File: .github/workflow/release.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          node-version: 16
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Build
+        run: ./scripts/build.sh
+      - name: Release
+        run: npx semantic-release@19
+        env:
+          GH_TOKEN: ${{secrets.GH_TOKEN}}
+          NPM_TOKEN: ${{secrets.NPM_TOKEN}}
```
Comment (TkDodo) [inline .github/workflow/release.yml:31]: i'm afraid I don't have the access rights to do so. @lstkz could you help us out here? 🙏

File: .github/workflow/release.yml (hunk 1 above)
Comment (vlad-yakovlev) [reply .github/workflow/release.yml]: Yep, but first we need change repo main branch from `master` to `main`

Comment (TkDodo) [reply .github/workflow/release.yml]: Right. I don't have permissions for that either

**Comment by vlad-yakovlev** [comment]
May need some fixes. I'll take a look tomorrow and give more info :)

**Comment by vlad-yakovlev** [comment]
No fixes required

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -41,3 +41,28 @@ jobs:
...
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Build
+        run: ./scripts/build.sh
+      - name: Release
+        run: npx semantic-release@19
```
Comment (TkDodo) [inline .github/workflows/ci.yml:65]:
```suggestion
        run: npx semantic-release@19 --dry-run
```

**Pull request merged**